File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
  var answer = 0;

  let w = 0;
  let h = 0;
  let maxw = 0;
  let maxh = 0;

  let squares = [];

  for (let i = 0; i < sizes.length; i++) {
    w = sizes[i][0];
    h = sizes[i][1];

    if (w < h) {
      squares.push([h, w]);
    } else {
      squares.push([w, h]);
    }
    if (maxw <= squares[i][0]) maxw = squares[i][0];
    if (maxh <= squares[i][1]) maxh = squares[i][1];
  }

  answer = maxw * maxh;

  return answer;
}

File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        int wMax = 0;
        int hMax = 0;
        
        for(int i = 0; i < sizes.length; i++) {
            // 가로의 길이가 무조건 세로 길이보다 길도록 자리 이동
            if(sizes[i][0] < sizes[i][1]) {
                int tmp = sizes[i][0];
                sizes[i][0] = sizes[i][1];
                sizes[i][1] = tmp;
            }
            // 가로와 세로의 최대 길이 찾기
            if(sizes[i][0] > wMax) wMax = sizes[i][0];
            if(sizes[i][1] > hMax) hMax = sizes[i][1];
        }
        /*
        for(int i = 0; i < sizes.length; i++) {
            System.out.println(sizes[i][0] + " " + sizes[i][1]);
        }
        */
        //System.out.println(wMax + " " + hMax);
        
        return wMax * hMax;
    }
}